.eslintrc

```diff
@@ -12,6 +12,7 @@
         "indent": 2,
         "prefer-const": 1,
         "react/jsx-curly-spacing": [ 2, "always" ],
-        "react/jsx-filename-extension": 0
+        "react/jsx-filename-extension": 0,
+        "semi": "off"
```

Comment:
I think semicolons are a matter of preferences, and just that. As long as you have a consistent style guide, you're good to go. The ECMA-262 clearly defined how the code is interpreted, so it's perfectly fine to omit semicolons.

Here is an extensive list of both sides of the argument, courtesy of @dtinth: https://goo.gl/iqoSo9

Personally, I'm not a fan of semicolons, and I omitted them in my codebases.

However, you have to be aware of a few edge cases. I strongly recommend adjusting your ESLint configuration to prevent edge cases, such as `semi`, `no-unexpected-multiline` and `no-unreachable`. You can take a look at how StandardJS handle these edge cases in their eslintrc.  

You can also use Prettier (prettier.io) to auto-format your code. Prettier works by traversing the AST, so it can detect common ASI misinterpretation and fix it for you.

Reply:
@dtinth Yes, i aware of some edge case that's why i want to asked that do we really aware of it. This is just question not blocking. Dev can go with thier passion :D

Reply:
At Agoda we're also push to no more semicolon too !!!